### DISCUSSION TITLE
Implementing dynamic dropdown in inspector and embed the selected value in cell object

### DISCUSSION THREAD

**Discussion by kvinita** [clientIO/joint#2572] [Q&A] [answered]
### Introduction

I'm currently working on a requirement where I have to create locations on paper and associate those locations to owners. When user drags a location on paper, the inspector must show a dropdown(or any other input type that suites this requirement) of all possible owners and user must be allowed to choose any one. Ideally, I want the chosen owner to be embedded in the cell object of the location as "ownerName". 

I get the list of owners from an API as they are dynamic. So I must bind the list of owners with the dropdown in inspector whenever an attempt to create new location is made on paper.

I have attached a sample for clarity:
![image](https://github.com/clientIO/joint/assets/124018900/2c4e86b0-b207-41b2-b9e4-7f7236148f00)

FYI : I have a table in database that stores location to owner mapping. Hence when user creates a new location on paper, he/she should be allowed to choose an owner so that I can insert the data in database accordingly. 

Would it be possible to implement this functionality? If there is any document/articles on it, kindly share that as well. Your assistance would be greatly appreciated. 
 

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kvinita** [comment, marked as answer]
We can create a select input type and provide dynamic options as follows:
```
owners: {
            type: 'select',
            options: cell.prop('owner'),
            multiple: false,
            overwrite: true,
            label:'Choose an owner',
            group: 'Owner',
            index:1
}
```

And, we can capture the chosen option by using an event as follows:
```
this.graph.on('change:owners', (element: joint.dia.Element)=>{
      const ownerName = element.get('owners');
    });
```

**Reply by kumilingus** [reply]
Note that the `options` can be defined as string, which means the path to an attribute.

```js
owners: {
    type: 'select',
    options: 'owners', // options are read from `owners` attribute
    multiple: false,
    overwrite: true,
    label:'Choose an owner',
    group: 'Owner',
    index:1
}
```
